File: c-sharp/palindromo.cs
using System;
using System.IO;

class Palindromo {
	public static void Main() {
		string texto = (new StreamReader(txt)).ReadLine("../string.txt");
		string palindromo = "";
		for(int i = 2; i < texto.Length; i++) {
			for(int j = 0; j < texto.Length; j++) {
				if(j + i <= texto.Length) {
					string fragmento = texto.Substring(j, i);
					if(CheckPalindromo(fragmento) && fragmento.Length > palindromo.Length) {
						palindromo = fragmento;
					}
				}
			}
		}
		Console.WriteLine(palindromo);
	}

	public static bool CheckPalindromo(String txt) {
		string mitadTexto = txt.Substring(0, txt.Length / 2);
		char[] arr = txt.ToCharArray();

		Array.Reverse(arr);
		string reversa = new string(arr);

		string mitadReversa = reversa.Substring(0, reversa.Length / 2);
		return mitadTexto.Equals(mitadReversa);
	}
}

File: js/palindromo.js
function palindromo() {
	fs = require('fs')
	fs.readFile('../string.txt', 'utf8', function (err,data) {
		if (err) { return console.log(err); }
		palindromo = "";

		for(i = 2; i < data.length; i++) {
			for(j = 0; j < data.length; j++) {
				if(j + i <= data.length) {
					fragmento = data.substring(j, j+i-1);
					if(checkPalindromo(fragmento) && fragmento.length > palindromo.length) {
						palindromo = fragmento;
					}
				}
			}	
		}
		console.info(palindromo);
	});
}

function checkPalindromo(txt) {
	reversa = txt.split("").reverse().join("");
	return txt == reversa;
}

palindromo();